Snowflake.Data/Core/Session/SFSessionHttpClientProperties.cs

```diff
@@ -1,78 +1,165 @@
 using System;
 using System.Collections.Generic;
-using System.Threading;
+using System.Linq;
+using Snowflake.Data.Client;
+using Snowflake.Data.Core.Session;
+using Snowflake.Data.Core.Tools;
 using Snowflake.Data.Log;
 
 namespace Snowflake.Data.Core
 {
 
     internal class SFSessionHttpClientProperties
     {
-        internal static readonly int s_maxHttpRetriesDefault = 7;
-        internal static readonly int s_retryTimeoutDefault = 300;
-        private static readonly SFLogger logger = SFLoggerFactory.GetLogger<SFSessionHttpClientProperties>();
-
+        private static readonly Extractor s_propertiesExtractor = new Extractor(new SFSessionHttpClientProxyProperties.Extractor());
+        public const int DefaultMaxPoolSize = 10;
```

Comment:
it is effectively internal because the class is internal. So it's matter of taste. I could change to internal if you want, but personally I don't feel the need. We could discuss it.